internal/handlers/pg/pgdb/query.go

```diff
@@ -268,46 +269,81 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 			}
 		}
 
-		// Handle _id with a simpler query, as it can't be an array
-		if k == "_id" {
-			switch v := v.(type) {
-			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
-				// type not supported for pushdown
-				// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-				// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
-			case float64, string, types.ObjectID, int32, int64:
-				// Select if value under the key is equal to provided value.
-				sql := `(_jsonb->%[1]s)::jsonb = %[2]s`
-
-				// placeholder p.Next() returns SQL argument references such as $1, $2 to prevent SQL injections.
-				// placeholder $1 is used for field key,
-				// placeholder $2 is used for field value v.
-				filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next()))
-				args = append(args, k, string(must.NotFail(pjson.MarshalSingleValue(v))))
-
-			default:
-				panic(fmt.Sprintf("Unexpected type of value: %v", v))
-			}
+		// check if values are equal or if the left value contains the right one
+		eqOperator := "@>"
 
-			continue
+		if k == "_id" {
+			// check if values are equal
+			eqOperator = "="
 		}
 
 		switch v := v.(type) {
-		case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
+		case *types.Document:
+			iter := v.Iterator()
+			defer iter.Close()
+
+			for {
+				docKey, docVal, err := iter.Next()
```

Comment:
In general, longer names should indicate larger scope. `k` and `v` are shorter, but more prominent. I think is is worth renaming them and those variables

Reply:
Yeah good point, will change that.